File: IT/1/Funzioni/Esercizi/Python/esercizio18.py
# file esercizio18.py
# -*- coding: utf-8 -*-

from __future__ import print_function, division
from random import randint

def recp(n):
	if n == 0: return float('nan')
	return 1/n

for i in range(5):
	r = randint(2, 15)
	print("Il reciproco di {} è {:.3f}.".format(r, recp(r)))

File: IT/1/Funzioni/Esercizi/Java/Esercizio18.java
/* file Esercizio18.java */

import java.util.Random;

public class Esercizio18 {
	static double recp(int n) {
		if (n == 0) return Double.NaN;
		return (double) 1/n;
	}

	public static void main(String [] args) {
		Random randomizer = new Random(System.currentTimeMillis());
		randomizer.nextInt();

		for (int i = 0; i < 5; i++) {
			int r = randomizer.nextInt(15) + 2;
			double rp = recp(r);
			System.out.println("Il reciproco di " + r + " è " + String.format("%.2f", rp) + ".");
		}
	}
}

